pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -196,10 +198,24 @@ func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOpera
 		}
 
 		var updated bool
-		for _, version := range operator.Status.Versions {
-			if version.Name == "operator" && version.Version == targetVersion {
-				updated = true
-				break
+		var mcoOperatorImageUpgrading bool
+		if operator.Name == "machine-config" {
+			for _, version := range operator.Status.Versions {
+				if version.Name == "operator-image" {
+					if targetImagePullSpec := getMCOImagePullSpec(mcoDeployment); targetImagePullSpec != version.Version {
```

Comment:
The `targetImagePullSpec := getMCOImagePullSpec(mcoDeployment)` can be computed just once, we do not need to do that inside every `for{ for{` iteration. So let's do it at the start of the function - and maybe even before that, and just pass the `mcoImage` inside as a string - which significantly its signature.

Reaching to MCO deployment is also quite a hack, so I think we should have a comment about why we do it this way.

Reply:
https://github.com/openshift/oc/pull/1924